### PR TITLE
New tooling for dependency management and building

### PR DESCRIPTION
**Pull request by Arno500**
I removed the following configurations:
- tox.ini
- requirements files
- setup.py

In favor of a unified build and dependency management system. Everything is configured inside the pyproject.toml and still accessible using the Makefile.
Dependencies are locked in standard using uv with the uv.lock. Everything will now be managed through uv, all at once with a single user interface and a single tool.
Please check the README for instructions

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,14 @@
-# ignore misc build/test files
-.env-lint
+# ignore uv venv
+.venv
+
+# ignore standard cache folders
+.*_cache
+
+# ignore built files
+hwbench.egg-info
```
Comment (anisse) [inline .gitignore:8]: Are you sure this is needed with uv build ?

Comment (Arno500) [reply .gitignore:8]: I had them in my local repo even working with uv, so I would say that they still can get into there

**Comment by anisse** [comment]
I think you forgot to remove setup.py, is that expected?

**Review by anisse — CHANGES_REQUESTED** [review]
Don't forget to remove setup.py

**Comment by Arno500** [comment]
Done!

**Review by anisse — APPROVED** [review]

**Pull request merged**